applications/ChimeraApplication/custom_processes/apply_chimera_process.h

```diff
@@ -0,0 +1,391 @@
+#include "omp.h"
+#include <unordered_map>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/linear_master_slave_constraint.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+#include "utilities/binbased_fast_point_locator.h"
+
+// Application includes
+#include "chimera_application_variables.h"
+#include "custom_utilities/distance_calcuation_utility.h"
+#include "custom_utilities/hole_cutting_utility.h"
+
+namespace Kratos {
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template <int TDim>
+class KRATOS_API(CHIMERA_APPLICATION) ApplyChimera : public Process {
+public:
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name Pointer Definitions
+    typedef Kratos::VariableComponent<Kratos::VectorComponentAdaptor<Kratos::array_1d<double, 3>>> VariableComponentType;
+    typedef ModelPart::IndexType IndexType;
+    typedef ModelPart::NodeType NodeType;
+    typedef Kratos::Variable<double> VariableType;
+    typedef std::vector<IndexType> ConstraintIdsVectorType;
+    typedef typename ModelPart::MasterSlaveConstraintType MasterSlaveConstraintType;
+    typedef typename ModelPart::MasterSlaveConstraintContainerType MasterSlaveConstraintContainerType;
+    typedef std::vector<MasterSlaveConstraintContainerType> MasterSlaveContainerVectorType;
+    typedef BinBasedFastPointLocator<TDim> PointLocatorType;
+    typedef typename PointLocatorType::Pointer PointLocatorPointerType;
+    typedef ModelPart::NodesContainerType NodesContainerType;
+
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyChimera);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Constructor
+     * @param rMainModelPart The reference to the modelpart which will be used
+     * for computations later on.
+     * @param iParameters The settings parameters.
+     */
+    explicit ApplyChimera(ModelPart& rMainModelPart, Parameters iParameters);
+
+    /// Destructor.
+    virtual ~ApplyChimera();
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetEchoLevel(int EchoLevel);
+
+    void SetReformulateEveryStep(bool Reformulate);
+
+    virtual void ExecuteInitializeSolutionStep() override;
+
+    virtual void ExecuteFinalizeSolutionStep() override;
+
+    virtual std::string Info() const override;
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    ModelPart& mrMainModelPart;
+    int mNumberOfLevels; 
+    Parameters mParameters;
```

Comment:
In your case I guess it is strictly needed to save the parameters as they contain the patch info (just commenting in case we can avoid it)

Reply:
Ya true.